src/views/token-swap/components/swap-form/swap-form.view.jsx

```diff
@@ -1,13 +1,92 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { BigNumber } from 'ethers'
+import { useLocation } from 'react-router-dom'
+
+import useSwapFormStyles from './swap-form.style'
+import { ReactComponent as ArrowDown } from '../../../../images/icons/arrow-down-circle.svg'
+import BoxAmount from '../amount-box/amount-box.view'
+
+function SwapForm ({
+  onGoToQuotes,
+  onOpenOfferSidenav,
+  accounts,
+  preferredCurrency,
+  fiatExchangeRates
+}) {
+  const classes = useSwapFormStyles()
+
+  const { search } = useLocation()
+  const urlSearchParams = new URLSearchParams(search)
+  const fromQuery = urlSearchParams.get('from')
+  const toQuery = urlSearchParams.get('to')
+
+  const [amount, setAmount] = React.useState(BigNumber.from(0))
+  const [tokensSelected, setTokens] = React.useState({})
+  const [whatDropdownIsActive, setDropdown] = React.useState('')
+
+  const setTokenPosition = tokenPosition => {
+    setTokens({ ...tokensSelected, ...tokenPosition })
+  }
+
+  React.useEffect(() => {
+    const from = accounts.find(a => a.accountIndex === fromQuery)
+    const to = accounts.find(a => a.accountIndex === toQuery)
+    setTokens({
+      from,
+      to
+    })
+  }, [accounts])
+
+  const switchTokens = () => {
+    setTokens({
+      from: tokensSelected.to,
+      to: tokensSelected.from
+    })
+  }
+
+  const renderBox = p => {
```

Comment:
what is `p`? better to use explicit names

Reply:
p is position